### PR TITLE
NXP-29450: Fix the conditions when DuplicatedCollectionListener will …

### PR DESCRIPTION
**Pull request by charlesboidot**
Fix for Jira Issue : https://jira.nuxeo.com/browse/NXP-29450

### CODE REVIEW

File: modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        collectionInFolder = session.createDocument(collectionInFolder);
+        session.save();
+        collectionManager.addToCollection(collectionInFolder, files, session);
+        Collection collectionAdapter = collectionInFolder.getAdapter(Collection.class);
+
+        //Check all children files are in the collection
+        assertEquals(NB_TEST_FILES,collectionAdapter.getCollectedDocumentIds().size());
+
+        // Create a new version of the folder
+        DocumentModel newVersionFolder = session.getDocument(session.checkIn(folder.getRef(), VersioningOption.MAJOR, ""));
```
Comment (tmartins) [inline modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java:90]: `newVersionFolder `never used, calling `checkIn` without reading the result is enough for this test

File: modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+@Features({ CoreFeature.class, CollectionFeature.class})
+public class CollectionCreateFolderVersionTest {
+
+    @Inject
+    CollectionManager collectionManager;
+
+    @Inject
+    CoreSession session;
+
+
```
Comment (tmartins) [inline modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java:53]: remove this line

File: modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class, CollectionFeature.class })
+public class CollectionCreateFolderVersionTest {
+
+    @Inject
+    CollectionManager collectionManager;
```
Comment (kevinleturc) [inline modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java:48]: Can you add protected keyword to fields not having visibility?

File: modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    @Inject
+    CoreSession session;
+
+    private static final String TEST_FOLDER = "testFolder";
+
+    private static final String TEST_COLLECTION_IN_FOLDER = "testCollectionInFolder";
+
+    protected static final String TEST_FILE_NAME = "testFile";
+
+    protected static final int NB_TEST_FILES = 3;
```
Comment (kevinleturc) [inline modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java:59]: Static fields should be first, everything should be protected.

File: modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        // Create a new version of the folder
+        session.checkIn(folder.getRef(), VersioningOption.MAJOR, "");
+        assertEquals(1, session.getVersions(folder.getRef()).size());
+
+        // Test if children documents are all still collected
+        DocumentModelIterator it = session.getChildrenIterator(folder.getRef());
+        int size = 0;
+        while (it.hasNext()) {
+            DocumentModel child = it.next();
+            if (child.getId() != collectionInFolder.getId()) {
```
Comment (kevinleturc) [inline modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java:96]: You should use the equals method, `=` in java checks references.
```suggestion
            if (child.getId().equals(collectionInFolder.getId())) {
```

File: modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        DocumentModelIterator it = session.getChildrenIterator(folder.getRef());
+        int size = 0;
+        while (it.hasNext()) {
+            DocumentModel child = it.next();
+            if (child.getId().equals(collectionInFolder.getId())) {
+                continue;
+            }
+            assertTrue(collectionManager.isCollected(child));
+            size++;
+        }
```
Comment (troger) [inline modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java:101]: Could it be simplified to:
```suggestion
        List<DocumentModel>children = session.getChildren(folder.getRef());
        int size = 0;
        for (DocumentModel child : children) {
            if (!child.getId().equals(collectionInFolder.getId())) {
                    assertTrue(collectionManager.isCollected(child));
                    size++;
            }
        }
```
?

File: modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        // Test if children documents are all still collected
+        List<DocumentModel>children = session.getChildren(folder.getRef());
+        int size = 0;
+        for (DocumentModel child : children) {
+            if (!child.getId().equals(collectionInFolder.getId())) {
+                    assertTrue(collectionManager.isCollected(child));
+                    size++;
+            }
+        }
+        assertEquals(NB_TEST_FILES, size);
```
Comment (kevinleturc) [inline modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java:100]: Could you format the code?

Comment (charlesboidot) [reply modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java]: Yes of course sorry

File: modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+        // Check all children files are in the collection
+        assertEquals(NB_TEST_FILES, collectionAdapter.getCollectedDocumentIds().size());
+
+        // Create a new version of the folder
+        session.checkIn(folder.getRef(), VersioningOption.MAJOR, "");
+        assertEquals(1, session.getVersions(folder.getRef()).size());
+
+        // Test if children documents are all still collected
+        List<DocumentModel>children = session.getChildren(folder.getRef());
```
Comment (kevinleturc) [inline modules/platform/nuxeo-collections/src/test/java/org/nuxeo/ecm/collections/core/test/CollectionCreateFolderVersionTest.java:92]: Missed format, as it is a new class you can run the formatter on the full file.

**Review by troger — APPROVED** [review]

**Review by kevinleturc — DISMISSED** [review]

**Comment by kevinleturc** [comment]
Please rebase your branch on master, it seems you're missing the new utests/server step.

Note: I aborted the previous build to take into account the next push.

**Review by kevinleturc — APPROVED** [review]

**Review by troger — APPROVED** [review]

**Pull request merged**